sunkit_image/coalignment/interface.py

```diff
@@ -0,0 +1,93 @@
+import warnings
+
+import numpy as np
+import sunpy.map
+from sunpy.util.exceptions import SunpyUserWarning
+
+from sunkit_image.coalignment.util.decorators import registered_methods
+
+__all__ = ["coalignment"]
+
+
+def convert_array_to_map(array_obj, map_obj):
+    """
+    Convert a 2D numpy array to a sunpy Map object using the header of a given
+    map object.
+
+    Parameters
+    ----------
+    array_obj : `numpy.ndarray`
+        The 2D numpy array to be converted.
+    map_obj : `sunpy.map.Map`
+        The map object whose header is to be used for the new map.
+
+    Returns
+    -------
+    `sunpy.map.Map`
+        A new sunpy map object with the data from `array_obj` and the header from `map_obj`.
+    """
+    header = map_obj.meta.copy()
+    header["crpix1"] -= array_obj.shape[1] / 2.0 - map_obj.data.shape[1] / 2.0
+    header["crpix2"] -= array_obj.shape[0] / 2.0 - map_obj.data.shape[0] / 2.0
+    return sunpy.map.Map(array_obj, header)
+
+
+def warn_user_of_nan(array, name):
```

Comment:
Within sunpy.util there is a warn_user function that you can use.

Reply:
Alright